New package: py-colorio

vivian-hafener opened the pull request.

vivian-hafener committed:
New package py-colorio

vivian-hafener committed:
New package py-colorio

manuelakuhn committed:
py-mako-1.1.6 (#29207)

* py-mako-1.1.6

* fix python dependency

adamjstewart commented on var/spack/repos/builtin/packages/py-colorio/package.py:
```diff
@@ -0,0 +1,22 @@
+
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-flit-core@3.2:3.7', type='build')
```
```suggestion
    depends_on('py-flit-core@3.2:3.6', type='build')
```
Version ranges are inclusive

adamjstewart commented on var/spack/repos/builtin/packages/py-colorio/package.py:
```diff
@@ -0,0 +1,22 @@
+    version('0.11.2', sha256='aa45d8e0a2e506c4019d4fb488d34a107d7f803c8e8ff355e2e57c01f6f1cd81')
+
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
```
Setuptools isn't needed, flit is used to build the project

vivian-hafener committed:
Update var/spack/repos/builtin/packages/py-colorio/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

vivian-hafener committed:
Update var/spack/repos/builtin/packages/py-colorio/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

vivian-hafener committed:
Fixed package

vivian-hafener committed:
Fixed package

becker33 committed:
rocmcc compiler: initial commit based on aocc and clang (#28575)

* rocmcc compiler: initial commit based on aocc and clang

Co-authored-by: luker <luke.roskop@hpe.com>
Co-authored-by: Tom Scogland <scogland1@llnl.gov>

zzzoom committed:
nss: add 3.75 (#29197)

iarspider committed:
New version: py-testpath 0.6.0; update py-flit-core dependency (#29200)

iarspider committed:
Add checksum for gmp 6.2.0 (#29212)

* Add checksum for gmp 6.2.0

* Update package.py

douglasjacobsen committed:
Revert GCS fetch strategy, to remove s3 interface (#28736)

This commit reverts the GCS fetch strategy to before commit:
d7596125231e800ca41c60e379be2b8abb47d20d

The previous commit added some s3 syntax to handle connections, but
added them into the GCS fetch strategy in a way that prevents GCS from
working anymore.

iarspider committed:
Add checksum for py-setuptools-scm 5.0.2 (#29213)

vivian-hafener committed:
New package: py-npx (#29324)

* New package py-npx

* Fixed linked issue

tkameyama committed:
llvm: fix build with Fujitsu compiler, and conflicts gcc@8.4 on aarch64 (#29030)

* llvm: fix build with Fujitsu compiler.

* Add conflict gcc 8.4 and aarch64.

* update conflicts version.

* fix version to apply patch.

eugeneswalker committed:
libdrm@2.4.101: needs meson for build (#29327)

natshineman committed:
Update MVAPICH2 to latest release verison 2.3.7 (#29329)

Co-authored-by: Nat Shineman <shineman.5@buckeyemail.osu.edu>

ax3l committed:
WarpX: 22.03 (#29334)

Update `warpx` & `py-warpx` to the latest release, `22.03`.

alalazo committed:
Fix importing Spack packages as Python modules (#29221)

fixes #29203

This PR fixes a subtle bug we have when importing
Spack packages as Python modules that can lead to
multiple module objects being created for the same
package.

It also fixes all the places in unit-tests where
"relying" on the old bug was crucial to have a new
"clean" state of the package class.

michaelkuhn committed:
qt: add 5.15.3 (#29328)

zzzoom committed:
mumps: add v5.4.1 (#29330)

vvolkl committed:
lhapdf: add python variant, change source tarball url to avoid python build failures (#29267)

climbfuji committed:
exempi: fix typo in "satisfies" condition (#29233)

nilsvu committed:
charmpp: resolve conflict with '+smp' (#29243)

The 'multicore' backend always uses SMP, so reverse
the logic of the `conflict` clause. This resolves an issue
where the '+smp' default caused the 'backend' to switch
away from 'multicore' unintentionally (#29234).

tkameyama committed:
boost: update version analysis patch for fujitsu compiler. (#29198)

wdconinc committed:
madx: new package, versions 5.07.00, 5.08.00, 5.08.01 (#28932)

sethrj committed:
slurm: add external find (#29146)

cessenat committed:
dos2unix: add the external find feature (#29335)

nilsvu committed:
Fix checksum for spectre@2022.02.17 (#29179)

haralmha committed:
edm4hep: Add dependencies to py-jinja2 and py-pyyaml (#28944)

liuyangzhuan committed:
butterflypack: add v2.1.1 (#29151)

MarieHouillon committed:
New version for openCARP packages, v9.0 (#29150)

Co-authored-by: openCARP consortium <info@opencarp.org>

Bambi committed:
kakoune: add new package (#29148)

enrico-usai committed:
aws-parallelcluster: add v2.11.5 (#29286)

vivian-hafener committed:
Merge branch 'py-colorio-actual' of https://github.com/viv-codes/spack into py-colorio-actual

vivian-hafener commented:
Rebased against upstream so that now it can build with it's dependency
